model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/connection/DefaultLiquibaseConnectionProvider.java

```diff
@@ -53,35 +45,8 @@
 
     private static final Logger logger = Logger.getLogger(DefaultLiquibaseConnectionProvider.class);
 
-    private static final AtomicBoolean INITIALIZED = new AtomicBoolean(false);
-
+    @SuppressWarnings("unused")
     public DefaultLiquibaseConnectionProvider(final KeycloakSession session) {
-        if (! INITIALIZED.get()) {
-            // TODO: all liquibase providers should probably synchronize on the same object.
-            synchronized (INITIALIZED) {
-                if (! INITIALIZED.get()) {
-                    initializeLiquibase();
-                    INITIALIZED.set(true);
-                }
-            }
-        }
```

Comment:
I can see that similar piece of code remains in `org.keycloak.connections.jpa.updater.liquibase.conn.DefaultLiquibaseConnectionProvider.create(session)`. I wonder why this change was removed here and not from the class above.

Reply:
In that class the initialization code remains because I had to make the provider initialize Liquibase's `Scope` object. Liquibase uses the thread context classloader by default and that was problematic in WildFly because the context classloader is the keycloak module classloader that offers no visibility to the Liquibase classes. [In the GHI](https://github.com/keycloak/keycloak/issues/9476) I've added some context about why this is needed.

Because the map storage is currently an optional feature I couldn't rely on the map-jpa `DefaultLiquibaseConnectionProvider` to do this initialization.